File: Juan/P1/S2/Napakalaki/src/napakalaki/BadConsequence.java
package napakalaki;

import java.util.ArrayList;

/**
 *
 * @author juane y Raúl
 * @version 1.1
 * Clase BadConsequence para el mal rollo del monstruo
 */
public class BadConsequence {
    /**
     * Atributos propios de la clase *
     */

    private String text; //lo que dice el mal royo
    private int levels; // niveles que se pierden
    private int nVisibleTreasures; // número de tesoros visibles que si pierden
    private int nHiddenTreasures; // número de tesoros ocultos que se pierden
    private boolean death; // representa un mal royo de tipo muerte

    /**
     * Atributos de tipo lista de TreasureKind *
     */
    private ArrayList<TreasureKind> specificHiddenTreasures = new ArrayList();
    private ArrayList<TreasureKind> specificVisibleTreasures = new ArrayList();

    /**
     * constructores 
     * Todos los atributos hay que inicializarlos
     */
    BadConsequence(String text, int levels, int nVisible, int nHidden) {
        this.text = text;
        this.levels = levels;
        this.nVisibleTreasures = nVisible;
        this.nHiddenTreasures = nHidden;
        this.death = false;
        this.specificVisibleTreasures = new ArrayList();
        this.specificVisibleTreasures = new ArrayList();
    }

    BadConsequence(String text, boolean death) {
        this.text = text;
        this.death = death;
        this.levels = 0;
        this.nVisibleTreasures = 10;
        this.nHiddenTreasures = 10; 
        this.specificVisibleTreasures = new ArrayList();        
        this.specificHiddenTreasures  = new ArrayList();
       
        
    }

    BadConsequence(String text, int levels, ArrayList<TreasureKind> tVisible,
            ArrayList<TreasureKind> tHidden) {
        this.text = text;
        this.levels = levels;
        this.nVisibleTreasures = 10; // valor por defecto que no va a a ser nunca así
        this.nHiddenTreasures = 10;
        this.death = false;
        this.specificVisibleTreasures = tVisible;
        this.specificHiddenTreasures = tHidden;
    }

    /** Consultores **/

    /**
     * Método getText() 
     * Devuelve lo que dice el mal royo
     *
     * @return text de tipo String
     */
    public String getText() {
        return this.text;
    }

    /**
     * Método getLevels() 
     * Devuelve los niveles que se pierden
     *
     * @return levels de tipo int
     */
    public int getLevels() {
        return this.levels;
    }

    /**
     * Método getnVisibleTreasures() 
     * Devuelve el número de tesoros visibles que
     * si pierden
     *
     * @return nVisibleTreasures de tipo int
     */
    public int getnVisibleTreasures() {
        return this.nVisibleTreasures;
    }

    /**
     * Método getnHiddenTreasures() 
     * Devuelve el número de tesoros ocultos que se
     * pierden
     *
     * @return nHiddenTreasures de tipo int
     */
    public int getnHiddenTreasures() {
        return this.nHiddenTreasures;
    }

    /**
     * Método isDeath() Devuelve true o false del mal royo de tipo muerte
     *
     * @return death de tipo boolean
     */
    public boolean isDeath() {
        return death;
    }

    /**
     * Método getSpecificHiddenTreasures()
     * Devuelve una lista con los Tesoros Ocultos
     * @return specificHiddenTreasures de tipo ArrayList
     */
    public ArrayList<TreasureKind> getSpecificHiddenTreasures() {
        return specificHiddenTreasures;
    }

    /**
     * Método getSpecificVisibleTreasures
     * Devuelve una lista con los Tesoros Visibles
     * @return specificVisibleTreasures de tipo ArrayList
     */
    public ArrayList<TreasureKind> getSpecificVisibleTreasures() {
        return specificVisibleTreasures;
    }

    /**
     * Método toString()
     * @return 
     */
    @Override
    public String toString() {
        
        if(specificVisibleTreasures.isEmpty()==false || specificHiddenTreasures.isEmpty()==false){
            return "BadConsequence = "  + text + ", levels = " + Integer.toString(levels) + ", nVisibleTreasures = " + specificVisibleTreasures
                + ", nHiddenTreasures = " +specificHiddenTreasures + ", Death = " + Boolean.toString(death) ;//
        }
        else
        {
           
            return "BadConsequence = " + text + ", levels = " + Integer.toString(levels) + ", nVisibleTreasures = " + Integer.toString(nVisibleTreasures) 
                + ", nHiddenTreasures = " + Integer.toString(nHiddenTreasures) + ", Death = " + Boolean.toString(death) ;//
        }
        
        
        
    }
}

File: Raul/P2/Ruby/Napakalaki/lib/BadConsequence.rb
#encoding: utf-8

#Versión 1.0
# author: juane, Raúl

require_relative 'Monster.rb'
require_relative 'TreasureKind.rb'
require_relative  'Prize.rb'
class BadConsequence

    #  @text #String que representa lo que dice el mal royo
    #  @levels # int para representar los niveles que se pierden
    #  @nVisibleTreasures # de tipo int, número de tesoros que se pierden
    #  @nHiddenTreasures # de tipo int, número de tesoros ocultos que se pierden
    #  @death # de tipo booleano, representa el mal rollo de tipo muerte
    #  
    #  #Para el tipo TreasureKind
    #  @specificHiddenTreasures
    #  @specificVisibleTreasures
  
  
    # Constructor #
    #LevelNumberOfTreasures

    def  initialize(aText,someLevels,someVisibleTreasures,someHiddenTreasures,
            someSpecificVisibleTreasures,someSpecificHiddenTreasures, death)
        @text = aText
        @levels = someLevels
        @nVisibleTreasures = someVisibleTreasures
        @nHiddenTreasures = someHiddenTreasures
        @specificVisibleTreasures = someSpecificVisibleTreasures
        @specificHiddenTreasures = someSpecificHiddenTreasures
        @death = death
   
    end 
    #Indica que el método new tiene visibilidad privada
    
    private_class_method :new

    def self.newLevelNumberOfTreasures(aText, someLevels,someVisibleTreasures,
            someHiddenTreasures)
        new(aText,someLevels,someVisibleTreasures,someHiddenTreasures,[],[],false)
    end
    
    def  self.newLevelSpecificTreasures (aText, someLevels,
            someSpecificVisibleTreasures, someSpecificHiddenTreasures)

        new(aText,someLevels,0,0,someSpecificVisibleTreasures,someSpecificHiddenTreasures,false)
    end
    
    def self.newDeath(aText)
        new(aText,0,0,0,[],[],true)
    end
   
#    Otra forma de definir los médotodos get y set 
#    #Método text
# 
#    def text
#        @text
#    end
#  
#    #Método levels
#    def levels
#        @levels
#    end
#  
#    #Método nvisible_treasures
#    def nvisible_treasures 
#        @nVisibleTreasures
#    end
#  
#    #Método nhidden_treasures
#    def nhidden_treasures
#        @nHiddenTreasures
#    end
#  
#    #Método death
#    def death
#        @death
#    end
#  
#    def specificHiddenTreasures
#        @specificHiddenTreasures
#    end
#    
#    def specificVisibleTreasure
#        @sepecificviVisibleTreasures
#    end
    
    #Método consultores de otra forma

    attr_accessor :text
    attr_accessor :levels
    attr_accessor :nVisibleTreasures
    attr_accessor :nHiddenTreasures
    attr_accessor :death
    attr_accessor :specificHiddenTreasures
    attr_accessor :specificVisibleTreasures

 
    def to_s 
        "BadConsequence =  #{@text}  , Levels =  #{@levels}  , nVisibleTreasure = #{@nVisibleTreasures}  nHiddenTreasures =  #{@nHiddenTreasures} , specificVisibleTreasures = #{@specificVisibleTreasures} , specicHiddenTreasures = #{@specificHiddenTreasures}, death =  #{@death}"     
    end  
  
   
end
